featurebyte/query_graph/node/function.py

```diff
@@ -0,0 +1,151 @@
+"""
+This module contains generic function related node classes
+"""
+# DO NOT include "from __future__ import annotations" as it will trigger issue for pydantic model nested definition
+from typing import Any, Dict, List, Literal, Optional, Sequence, Tuple
+
+from pydantic import BaseModel, Field
+
+from featurebyte.enum import DBVarType, FunctionParameterInputForm
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.query_graph.enum import NodeOutputType, NodeType
+from featurebyte.query_graph.node.base import BaseSeriesOutputNode
+from featurebyte.query_graph.node.metadata.operation import (
+    DerivedDataColumn,
+    NodeOutputCategory,
+    OperationStructure,
+    OperationStructureBranchState,
+    OperationStructureInfo,
+    PostAggregationColumn,
+)
+from featurebyte.query_graph.node.metadata.sdk_code import (
+    CodeGenerationConfig,
+    CodeGenerationContext,
+    ExpressionStr,
+    StatementT,
+    ValueStr,
+    VariableNameGenerator,
+    VarNameExpressionInfo,
+)
+
+
+class FunctionParameterInput(BaseModel):
```

Comment:
does it make sense to have something like 

```
class BaseFunctionParameter:
  dtype: DBVarType
  def get_column_args -> Optional[str]:
    return None
  def get_literal_value -> expressions.Expression:
    # get each implementation to override
    pass

class ValueFunctionParameter(BaseFunctionParameter)
  value: Any

class ColumnFunctionParameter(BaseFunctionParameter)
  column: str
  def get_column_args -> Optional[str]:
    return self.column
```

this way, we should be able to skip all the if-else checks on the `input_form` type

Reply:
Good point. If i understand this correctly, basically we will have different function parameter: column type & value type. Will update this accordingly.